modules/internal/ChapelArray.chpl

```diff
@@ -2037,7 +2061,18 @@ module ChapelArray {
             hi = this.domain.high+1;
       const newDom = {lo..hi};
       on this._value {
-        this._value.dsiReallocate(newDom);
```

Comment:
While there - I would prefer 'newDom' to be a range. That will make the generate code slimmer+faster.
I do not see the need for 'newDom' to be a domain.

This could be done as a separate PR.